### PR TITLE
Signin

### PR DESCRIPTION
**Pull request by kiranpratyush**
PR for sign-in  page
- Provided firebase authentication

https://user-images.githubusercontent.com/77445957/161500319-95708b87-cab4-47d0-93e6-1ed680525886.mp4

### CODE REVIEW

File: public/style.css [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  background-color: aliceblue;
+  width: 8px;
+}
+::-webkit-scrollbar-thumb {
+  border-radius: 1000px;
+  background-color: rgb(83, 79, 79);
+  border: 2px solid #111;
+}
+::-webkit-scrollbar-thumb:hover {
+  background-color: rgb(170, 177, 177);
```
Comment (mimansha-swarup) [inline public/style.css]: better use variables

File: src/Components/Header/Header.css [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  font-weight: 500;
+}
+
+.first-nav {
+  display: flex;
+  justify-content: space-evenly;
+  gap: 10px;
+}
+/* .first-nav a {
+  color: var(--color-black);
```
Comment (mimansha-swarup) [inline src/Components/Header/Header.css]: remove unrequired comments

File: src/Components/SignUp/SignUp.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  const nameRef = useRef();
+  const emailRef = useRef();
+  const passwordRef = useRef();
+  const confirmPasswordRef = useRef();
+  const navigate = useNavigate();
+  function handleSubmit(e) {
+    e.preventDefault();
+    const email = emailRef.current.value;
+    const password = passwordRef.current.value;
+    const confirmPassword = confirmPasswordRef.current.value;
```
Comment (mimansha-swarup) [inline src/Components/SignUp/SignUp.js:19]: wow this is great approach

File: src/Pages/SignInPage/SignInPage.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { SignIn } from '../../Components/SignIn/SignIn';
+import { SignUp } from '../../Components/SignUp/SignUp';
+import { Header } from '../../Components/Header/Header';
```
Comment (mimansha-swarup) [inline src/Pages/SignInPage/SignInPage.js:4]: suggestions to use barrel  export

**Review by mimansha-swarup — APPROVED** [review]

File: src/App.js [hunk 5]
```diff
@@ -1,10 +1,33 @@
...
-    </div>
-}
\ No newline at end of file
+import React from 'react';
+import { Routes, Route } from 'react-router-dom';
+import HomePage from './Pages/HomePage';
+import ShopPage from './Pages/ShopPage';
+import { SignInPage } from './Pages/SignInPage/SignInPage';
+import { CartPage } from './Pages/CartPage';
+import { WishListPage } from './Pages/WishListPage';
```
Comment (manojasarsa) [inline src/App.js]: You can make use of barrel export.

File: src/fireBaseAuth.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  }
+}
+async function signIn(email, password) {
+  try {
+    const user = await signInWithEmailAndPassword(auth, email, password);
+    console.log(user.user.uid)
+    const userName = user.user.uid;
+    return userName;
+  } catch (error) {
+    console.log(error)
```
Comment (manojasarsa) [inline src/fireBaseAuth.js:25]: You can use `console.error` for handling errors.

**Review by manojasarsa — APPROVED** [review]
Rest, code is pretty consistent and clean.
Good to go.

File: src/App.js (hunk 5 above)
Comment (kiranpratyush) [reply src/App.js]: Done Waiting to merge from other branch

File: src/fireBaseAuth.js (hunk 6 above)
Comment (kiranpratyush) [reply src/fireBaseAuth.js:25]: Yes thank you

File: src/Pages/SignInPage/SignInPage.js (hunk 4 above)
Comment (kiranpratyush) [reply src/Pages/SignInPage/SignInPage.js:4]: Done waiting to merge from other branch

File: src/Components/Header/Header.css (hunk 2 above)
Comment (kiranpratyush) [reply src/Components/Header/Header.css]: removed

File: public/style.css (hunk 1 above)
Comment (kiranpratyush) [reply public/style.css]: sure thank you

**Pull request merged**